hazelcast/src/main/java/com/hazelcast/datastore/CloseableDataSource.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.datastore;
+
+import com.hazelcast.spi.annotation.Beta;
+
+import javax.sql.DataSource;
+import java.io.PrintWriter;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.SQLFeatureNotSupportedException;
+import java.util.logging.Logger;
+
+/**
+ * Wrapper of the datasource which handles the closing of the non-shared datasources
+ *
+ * @since 5.2
+ */
+@Beta
+public abstract class CloseableDataSource implements DataSource, AutoCloseable {
```

Comment:
is this class part of public api? if not better to put it under some impl package.